src/Stratis.Bitcoin.Features.MemoryPool/MempoolValidator.cs

```diff
@@ -438,6 +438,19 @@ private async Task AcceptToMemoryPoolWorkerAsync(MempoolValidationState state, T
                     return;
                 }
 
+                // Dont accept transactions to the memory pool that are already known on the valid chain.
+                FetchCoinsResponse fetchCoinsResponse = this.coinView.FetchCoins(new[] { tx.GetHash() });
+                if (fetchCoinsResponse?.BlockHash != null)
+                {
+                    ChainedHeader header = this.chainIndexer.GetHeader(fetchCoinsResponse.BlockHash);
+                    if (header != null && header.IsAssumedValid)
```

Comment:
I don't think `IsAssumedValid` gets set for every header, only those that fall under the `-assumevalid` CLI option, or are below the last checkpoint.